battenberg_gitlab/utils.py

```diff
@@ -0,0 +1,36 @@
+import os
+import logging
+import tempfile
+from gitlab import Gitlab
+from pygit2 import clone_repository, discover_repository, Keypair, RemoteCallbacks, Repository
+from battenberg import construct_keypair
+
+
+logger = logging.getLogger(__name__)
+
+
+def init_gitlab(config_file: str, gitlab_server: str) -> Gitlab:
+    if not config_file:
+        config_file = os.path.abspath(os.path.join(os.path.dirname(__file__), '..',
+                                                   'python-gitlab.cfg'))
+    logger.debug(f'Reading gitlab config from {config_file}')
+    return Gitlab.from_config(gitlab_server, [config_file])
+
+
+def ensure_workspace(workspace_path: str = None) -> str:
+    if not workspace_path:
+        workspace_path = tempfile.mkdtemp()
+    elif not os.path.exists(workspace_path):
+        os.makedirs(workspace_path)
+    return workspace_path
+
+
+def clone_or_discover_repo(repo_url: str, local_path: str) -> Repository:
+    try:
+        return Repository(discover_repository(local_path))
+    except Exception:
```

Comment:
I'm surprised this exception is caught and not raised. Since you're searching for repos, I imagine that they should all exist unless one is deleted between the time you searched and when you attempt to clone it. If there is something else you attempting to solve by catching the exception, maybe a comment would help clarify.

Reply:
I've added a comment here to clarify why we're catching the exception.